crates/steel/src/host/db/proof.rs

```diff
@@ -311,3 +366,53 @@ fn add_proof(
 
     Ok(())
 }
+
+/// Converts an API ReceiptResponse into a vector of ReceiptEnvelope.
+fn convert_rpc_receipts<N: Network>(
+    rpc_receipts: impl IntoIterator<Item = <N as Network>::ReceiptResponse>,
+    receipts_root: B256,
+) -> Result<Vec<ReceiptEnvelope>> {
+    let receipts = rpc_receipts
+        .into_iter()
+        .map(|rpc_receipt| {
+            // Unfortunately ReceiptResponse does not implement ReceiptEnvelope, so we have to
+            // manually convert it. We convert to a TransactionReceipt which is the default and
+            // works for Ethereum-compatible networks.
+            // Use serde here for the conversion as it is much safer than mem::transmute.
+            // TODO(https://github.com/alloy-rs/alloy/issues/854): use ReceiptEnvelope directly
+            let json = serde_json::to_value(rpc_receipt).context("failed to serialize")?;
+            let tx_receipt: TransactionReceipt = serde_json::from_value(json)
+                .context("failed to parse as Ethereum transaction receipt")?;
+
+            Ok(tx_receipt_to_envelope(tx_receipt))
```

Comment:
It is very unfortunate that in the current version of Alloy a ReceiptResponse does not implement the ReceiptEnvelop (as the TransactionResponse does for example).
So we have to convert it manually. I tried two approaches:
- Switch from the general `Network` to `Ethereum`, where the trait boundaries are correct. However, this completely changes the bounds in Steel and breaks op-steel support.
- Cast" the receipt response. The current code does this in a memorysafe way with serde-json. This is ugly, but at least it works for everything that is Ethereum compatible. (It will not otherwise cause any weird bugs in the guest, as the resulting receipt root is verified.)

Eventually, this can only be fixed when https://github.com/alloy-rs/alloy/issues/854 is fixed upstream.